src/tribler/gui/tribler_window.py

```diff
@@ -681,13 +679,16 @@ def on_create_torrent_updates(self, update_dict):
 
     def on_add_torrent_browse_file(self, *_):
         self.raise_window()  # For the case when the action is triggered by tray icon
-        filenames = QFileDialog.getOpenFileNames(
+        filenames, *_ = QFileDialog.getOpenFileNames(
             self, tr("Please select the .torrent file"), QDir.homePath(), tr("Torrent files%s") % " (*.torrent)"
         )
-        if len(filenames[0]) > 0:
-            for filename in filenames[0]:
-                self.pending_uri_requests.append(Path(filename).as_uri())
-            self.process_uri_request()
+        if not filenames:
+            return
+
+        for filename in filenames:
+            uri = Path(filename).resolve().as_uri()
```

Comment:
I don't see any harm in using resolve() here but I wonder if this will fix the issue since the expected file is .torrent files which should exist as it is selected using the file dialog. The reproduction using `['.']` might not be representative enough for this issue.

Reply:
Changing the test from the original to the following produces the same reproducibility.

```python
def test_on_add_torrent_browse_file(tribler_window: TriblerWindow):
    """ Test that the on_add_torrent_browse_file method works correctly"""
    tribler_window.raise_window = Mock()
    tribler_window.process_uri_request = Mock()

    with patch.object(QFileDialog, 'getOpenFileNames', Mock(return_value=['./any.torrent'])) as patched_getOpenFileNames:
        tribler_window.on_add_torrent_browse_file()

    assert tribler_window.raise_window.called
    assert patched_getOpenFileNames.called
    assert tribler_window.process_uri_request.called
```